scripts/django_tests_settings.py

```diff
@@ -9,3 +9,5 @@
     "django.contrib.sessions",
     "django.contrib.sites",
 ]
+
+STATIC_URL = "static/"
```

Comment:
Is it necessary to set the `STATIC_URL` setting?

Reply:
Yes. If there is no `STATIC_URL` setting, the following error occurs during stubtest.

```sh
error: django.contrib.staticfiles.management.commands.collectstatic encountered unexpected error, ImproperlyConfigured: You're using the staticfiles app without having set the required STATIC_URL setting.
Stub: in file /app/django-stubs/contrib/staticfiles/management/commands/collectstatic.pyi:1
N/A
Runtime: in file /usr/local/lib/python3.12/site-packages/django/contrib/staticfiles/management/commands/collectstatic.py
This is most likely the fault of something very dynamic in your library. It's also possible this is a bug in stubtest.
If in doubt, please open an issue at https://github.com/python/mypy

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/django/core/files/storage/handler.py", line 35, in __getitem__
    return self._storages[alias]
           ~~~~~~~~~~~~~~^^^^^^^
KeyError: 'staticfiles'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/mypy/stubtest.py", line 247, in test_module
    yield from verify(stub, runtime, [module_name])
  File "/usr/local/lib/python3.12/site-packages/mypy/stubtest.py", line 404, in verify_mypyfile
    and _belongs_to_runtime(runtime, m)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/mypy/stubtest.py", line 364, in _belongs_to_runtime
    if isinstance(obj, types.ModuleType):
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/utils/functional.py", line 280, in __getattribute__
    value = super().__getattribute__(name)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/utils/functional.py", line 251, in inner
    self._setup()
  File "/usr/local/lib/python3.12/site-packages/django/contrib/staticfiles/storage.py", line 540, in _setup
    self._wrapped = storages[STATICFILES_STORAGE_ALIAS]
                    ~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/core/files/storage/handler.py", line 43, in __getitem__
    storage = self.create_storage(params)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/core/files/storage/handler.py", line 55, in create_storage
    return storage_cls(**options)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/django/contrib/staticfiles/storage.py", line 29, in __init__
    check_settings(base_url)
  File "/usr/local/lib/python3.12/site-packages/django/contrib/staticfiles/utils.py", line 49, in check_settings
    raise ImproperlyConfigured(
django.core.exceptions.ImproperlyConfigured: You're using the staticfiles app without having set the required STATIC_URL setting.
```

Reply:
Aha, interesting. Good find